clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryReporter.java

```diff
@@ -623,14 +622,7 @@ public void initiateClose(long timeoutMs) {
                     return;
                 }
 
-                try {
-                    log.info("About to wait {} ms. for terminal telemetry push to be submitted", timeoutMs);
-                    if (!terminalPushInProgress.await(timeoutMs, TimeUnit.MILLISECONDS)) {
```

Comment:
this makes total sense to me basically because we cannot ensure that there will always be a call to `networkClient.poll` when closing a consumer, and here we were relying on that call to transition from `TERMINATING_PUSH_NEEDED` to `TERMINATING_PUSH_IN_PROGRESS` on [networkClient.poll](https://github.com/apache/kafka/blob/c36b993af0f79a7d3be773698fe4a6d3d0169888/clients/src/main/java/org/apache/kafka/clients/NetworkClient.java#L1318), so we could definitely get stuck for the timeout.

Reply:
Great, so are we good with the PR?